gammapy/irf/io.py

```diff
@@ -124,3 +129,49 @@ def load_cta_irfs(filename):
     psf = EnergyDependentMultiGaussPSF.read(filename, hdu="POINT SPREAD FUNCTION")
 
     return dict(aeff=aeff, bkg=bkg, edisp=edisp, psf=psf)
+
+
+def load_irf_dict_from_file(filename):
+    """Open a fits file and generate a dictionary containing the Gammapy objects
+    corresponding ot the IRF components stored
+    
+    Parameters
+    ----------
+    filename : str, Path
+        path to the file containing the IRF components, if EVENTS and GTI HDUs 
+        are included in the file, they are ignored
+
+    Returns
+    -------
+    irf_dict : dict
+        dictionary with instances of the Gammapy obejcts corresponding 
+        to the IRF components        
+    """
+    filename = make_path(filename)
+
+    hdulist = fits.open(make_path(filename))
+    
+    irf_dict = {}
+
+    for hdu in hdulist:
+        hdu_class = hdu.header.get("HDUCLAS1", "").lower()
+        
+        if hdu_class == "response":
+            hdu_class = hdu.header.get("HDUCLAS4", "").lower()
+        
+            loc = HDULocation(
+                hdu_class=hdu_class,
+                hdu_name=hdu.name,
+                file_dir=filename.parent,
+                file_name=filename.name
+            )
+            
+            for name in HDUIndexTable.VALID_HDU_TYPE:
```

Comment:
Why do we need to loop here? Why not just check `if hdu_class in HDUIndexTable.VALID_HDU_TYPE`?

Reply:
Also, the `VALID_HDU_TYPE` and `VALID_HDU_CLASS` should probably be sets, not lists

Reply:
The problem that is solved here is to assign the class instances to the corresponding observation attributes (and attribute names). That way it works here (basically accidentally, but safe...) is that the hdu class (`bkg_2d`, `aeff_2d` etc.) contains the attribute name. The alternative solution here is something like:
```python3
irf = loc.load()

if instance(irf, PSF):
    kwargs["psf"] = irf
elif instance(irf, EffectiveArea2D):
    kwargs["aeff"] = irf
...
```
Or even better introducing a Gammapy internal`IRF.type` class attribute (see TODO in the code), such that the following works:
```python3
irf = loc.load()
kwargs[irf.type] = irf
```
As I mentioned before for `Map` objects we have a factory pattern, where `Map.read()` returns the given instance of `WcsNDMap`, `HpxNDMap` or `RegionNDMap`. Something like this could work for IRFs as well, like:

```python3
try:
    kwargs["psf"] =  PSF.read() # find first valid HDU here...
except ValueError:
    pass
```
But the latter I'd consider as work for a follow up PR.